baseplate/file_watcher.py

```diff
@@ -61,14 +63,34 @@ class FileWatcher(object):
     :param callable parser: A callable that takes an open file object, parses
         or otherwise interprets the file, and returns whatever data is
         meaningful.
+    :param float timeout: How long, in seconds, to block instantiation waiting
+        for the watched file to become available (defaults to not blocking).
 
     """
-    def __init__(self, path, parser):
+    def __init__(self, path, parser, timeout=None):
```

Comment:
Should this be a `Timedelta`? That seems like a somewhat common pattern in baseplate.

Reply:
Also, then it’s easy to just use the timespan config option for this rather than passing it to the `from_config` call

Reply:
The reason I put it on the `from_config` call instead of in config itself is that I think it's the sort of thing the programmer will know at the call-site rather than the application configuration. You know you want a specific line of code to block so that subsequent lines can safely trust the data's there; a config file doesn't really have the same context.